boa/src/builtins/string/mod.rs

```diff
@@ -722,7 +704,12 @@ impl String {
             None
         };
 
-        Self::string_pad(primitive_val, max_length, fill_string, false)
+        Self::string_pad(
+            primitive_val.to_string(),
+            max_length,
+            fill_string.map(|x| x.to_string()),
```

Comment:
I think we can just pass the `to_string()` function to the `map()` here, right? Instead of creating the closure. Passing `Rc::to_string` (if I'm not mistaken on what this is xD)

Reply:
Yes. we can. but I think we can make this even better by accepting a `RcString` instead of `StdString` this will eliminate a `String::clone`, also we can clean it even more if we pass in a `RcString` direclty instead of `Option<RcString>`, because the will string is defaulted if it's `None` to `" "` also in `pad_string` it makes more sense for the fill string that we are padding to be specified.